src/main/java/org/graylog/integrations/aws/service/KinesisService.java

```diff
@@ -128,9 +127,7 @@ public KinesisHealthCheckResponse healthCheck(KinesisHealthCheckRequest request)
         // Retrieve one records from the Kinesis stream
         final List<Record> records = retrieveRecords(request.streamName(), kinesisClient);
         if (records.size() == 0) {
-            String explanation = "The Kinesis stream does not contain any messages.";
-            LOG.error(explanation);
-            return KinesisHealthCheckResponse.createFailed(explanation);
+            throw new BadRequestException("The Kinesis stream does not contain any messages.");
```

Comment:
Would be helpful to include the stream name. 
throw new BadRequestException(String.format("The Kinesis stream [%s] does not contain any messages.", request.streamName()));

Reply:
Great idea! I'll fix.